### DISCUSSION TITLE
Is there any possibility for spawning custom peds?

### DISCUSSION THREAD

**Discussion by patryksuu** [scripthookvdotnet/scripthookvdotnet#1515] [Q&A] [answered]
can I spawn addon ped in v2? source/scripting_v2/GTA.Native/PedHash.cs

is there option to spawn by model name?

**Comment by kagikn** [comment]
well, you can spawn a ped with any `Model`, like `World.CreatePed("player_zero", Game.Player.Character.Position)` and `World.CreatePed((Model)unchecked((int)0x9b22dbaf), Game.Player.Character.Position)`. `Model` defines implicit operators that convert an `int` and a `string` into a `Model`.

**Reply by JohnFromGWN** [reply, marked as answer]
i think by "custom ped" @patryksuu is not referring to peds like player_zero which is as vanilla as it gets. He means mods. So, while this is a late answer, there are dozens of different ways to indeed spawn custom peds (replace or addons) and an infinite number of combinations. However v2 is no longer supported so v3 should be the way to go. Here is an example for any noob coming across this discussion.

```cs
// Create the Ped
Ped ped;

// Determine if the identifier is a string (name) or an enumeration value (PedHash)
 if (identifier is string pedName)
{
     ped = World.CreatePed(pedName, spawnPosition);
}
else if (identifier is PedHash pedHash)
{
     ped = World.CreatePed(pedHash, spawnPosition);
}
else
{
    throw new ArgumentException("Identifier must be a string (name) or PedHash.");
}
```

**Reply by kagikn** [reply]
i thought my examples explained how to spawn peds with any models good enough, but it appears that @patryksuu thought the snippet provided by @JohnFromGWN explained better.

i admit i explained worse this time, i could've explained with non-vanilla model hashes, though it would've been a lot better if @patryksuu told me my snippets hadn't made enough sense to them 🥲

**Reply by JohnFromGWN** [reply]
I don't think there was anything wrong with your answer, I think it was just the question from the OP was more about addon peds. I know you're a much more skilled programmer than I am, or anyone else in the GTA V community, but I'm not sure about your second example. I tried it for fun and it did not work.  From my experience, these are the ways to spawn peds, but i'm sure there are more.

//Example of addon mod Kira
MyPed = World.CreatePed("Kira", PP.Position + PP.ForwardVector * 3f, PP.Heading + 90);

// 4 equivalent examples of vanilla mod Vagos female
MyPed = World.CreatePed(new Model(1520708641),PP.Position + PP.ForwardVector * 3f, PP.Heading + 90); 
MyPed = World.CreatePed("g_f_y_vagos_01", PP.Position + PP.ForwardVector * 3f, PP.Heading + 90); 
MyPed = World.CreatePed(PedHash.Vagos01GFY,PP.Position + PP.ForwardVector * 3f, PP.Heading + 90);
MyPed = World.CreatePed(new Model(0x5AA42C21), PP.Position + PP.ForwardVector * 3f, PP.Heading + 90);

// Examples of Franklin vanilla player
MyPed = World.CreatePed("player_one", PP.Position + PP.ForwardVector * 3f, PP.Heading + 90);
MyPed = World.CreatePed(PedHash.Franklin, PP.Position + PP.ForwardVector * 3f, PP.Heading + 90);

**Reply by kagikn** [reply]
Oops, my second example was wrong about how to cast literal integers to another integer type that can't exactly represent the target values, where an `unchecked` expression was required. Yes, you could just directly use `new Model(int)`, though you would have to use an `unchecked` expression when you cast a literal `int` that can't exactly represent the value as `uint` to `uint`.

Also, thank you for cheering me up.
